verifier/src/main/java/com/aws/jverify/verifier/compiler/dafnygenerator/base/ExpressionCompiler.java

```
@@ -557,14 +580,34 @@ private Expression translateFieldAccess(JCTree.JCFieldAccess fieldAccess, IOrigi
             // Ignore package qualification
             return new NameSegment(origin, baseGenerator.nameCompiler.getCompiledName(classSymbol, origin), List.of());
         }
+
+        if (fieldAccess.sym instanceof Symbol.VarSymbol varSymbol &&
+            varSymbol.owner instanceof Symbol.ClassSymbol ownerClass &&
+            (ownerClass.fullname.contentEquals(Double.class.getName()) ||
+             ownerClass.fullname.contentEquals(Double.class.getName() + "?static"))) {
+            var fieldNameStr = fieldAccess.name.toString();
+            return switch (fieldNameStr) {
+                case "NaN" -> fp64Constant(origin, "NaN");
+                case "POSITIVE_INFINITY" -> fp64Constant(origin, "PositiveInfinity");
+                case "NEGATIVE_INFINITY" -> fp64Constant(origin, "NegativeInfinity");
+                case "MAX_VALUE" -> fp64Constant(origin, "MaxValue");
+                case "MIN_VALUE" -> fp64Constant(origin, "MinSubnormal");
+                case "MIN_NORMAL" -> fp64Constant(origin, "MinNormal");
+                default -> {
+                    reporter.reportError(fieldAccess, "notSupported", "Double field " + fieldNameStr);
```

What are the not supported fields? Could you add tests for those? Also, are you sure that we should report an error for them? Can't users still add contracts for them?

---

Removed the error-reporting default case. Now only handles the 6 fp64 constants (NaN, POSITIVE_INFINITY, NEGATIVE_INFINITY, MAX_VALUE, MIN_VALUE, MIN_NORMAL). Other fields fall through to normal handling.